flatsurf/geometry/homology.py

```diff
@@ -361,6 +377,64 @@ def coefficient(self, gen):
 
         return self.coefficients()[index]
 
+    @cached_method
+    def simplex_coefficients(self):
+        r"""
+        Return pairs (coefficient, simplex) such that this homology class can
+        be written as the chain Σ coefficient · simplex.
+
+        EXAMPLES::
+
+            sage: from flatsurf import translation_surfaces, SimplicialHomology
+            sage: T = translation_surfaces.square_torus()
+            sage: H = SimplicialHomology(T)
+            sage: a, b = H.gens()
+            sage: (a - b).simplex_coefficients()
+            ((1, (0, 1)), (-1, (0, 0)))
+
+        """
+        return tuple((self._chain[s], s) for s in self._chain.support())
+
+    @cached_method
+    def holonomy(self):
```

Comment:
I do not think it is a good idea to have this at the level of homology. More precisely, homology makes sense for a topological surface (even though right now it belongs to the similarity surface category). But the holonomy is only valid for translation surfaces (which is a sub category of similarity surfaces).

What would be more natural is `surface.holonomy_morphism(domain)`.

Reply:
Ok. Then I'll just add it as an example to `chain()`.